### PR TITLE
Lightware and Maxbotix I2C: search buses 0 and 1

### LINKED ISSUE #5831: RangeFinders using I2C (Maxbotix I2C, LightwareI2C) should be supported on 2nd I2C bus for Pixhawk2
#### Issue details
Have multiple airframes using Maxbotix I2C EZ4 sonars for auto landing.  Just converted to PH2.1 and am no longer getting any reading from the rangefinder.  Other I2C items (airspeed) work just fine, so I assume this is a software problem.  Have tried multiple planes and multiple sonar units.

#### Version
PH 2.1
AP 3.7.1
MP 1.3.44

#### Platform
[X] All
[  ] AntennaTracker
[  ] Copter
[  ] Plane
[  ] Rover

#### Airframe type
Multiple...pusher flying wing (X8), custom puller aircraft, and others

#### Hardware type
PH2.1

### PR DESCRIPTION
**Pull request by OXINARF**
After three PRs trying to fix Lightware detection in the Cube (#6558, #6710 and #6991) in the wrong way, I decided to do it myself with the best solution we have now - search buses 0 and 1 when a specific bus isn't specified. Hopefully we'll get to a point where we can get a list of I2C buses to search and we remove the macro.

@chobitsfan Can you please check this works for you?
@rmackay9 @mirkix If either of you can test on a BeagleBone board it would be great. I'm pretty sure I didn't break it, but I don't have hardware to test.

Also made a commit to search both buses for Maxbotix I2C.
All these should fix #5831.

### CODE REVIEW

**Comment by chobitsfan** [comment]
@OXINARF OK. I will test it once I back to lab tomorrow.

**Comment by chobitsfan** [comment]
I have tested it with lightware i2c rangefinder sf20. It works well. Thank you very much

**Comment by mirkix** [comment]
@OXINARF My SF11 has a broken connector, I can not test is PR at the moment.

**Comment by n1kitap** [comment]
Hello, I had hand-made board of Pixhawk 1, and I had to remap CAN pins to I2C1 (in stm32f4 board config), so I had only I2C2 bus active. Lightware SF11 succesfully work with this patch.

**Comment by OXINARF** [comment]
@chobitsfan @n1kitap Thank you for your testing.

@mirkix That's unfortunate. I'm confident enough to merge this without your testing, so feel free to smack me in the head if I mess it up 🙂

File: libraries/AP_RangeFinder/AP_RangeFinder_MaxsonarI2CXL.cpp [hunk 1]
```diff
@@ -46,10 +47,11 @@ AP_RangeFinder_MaxsonarI2CXL::AP_RangeFinder_MaxsonarI2CXL(RangeFinder::RangeFin
    trying to take a reading on I2C. If we get a result the sensor is
    there.
 */
-AP_RangeFinder_Backend *AP_RangeFinder_MaxsonarI2CXL::detect(RangeFinder::RangeFinder_State &_state)
+AP_RangeFinder_Backend *AP_RangeFinder_MaxsonarI2CXL::detect(RangeFinder::RangeFinder_State &_state,
+                                                             AP_HAL::OwnPtr<AP_HAL::I2CDevice> dev)
 {
     AP_RangeFinder_MaxsonarI2CXL *sensor
-        = new AP_RangeFinder_MaxsonarI2CXL(_state);
+        = new AP_RangeFinder_MaxsonarI2CXL(_state, std::move(dev));
```
Comment (gmorph) [inline libraries/AP_RangeFinder/AP_RangeFinder_MaxsonarI2CXL.cpp]: Sorry to comment on an already merged PR but I don't understand why we are calling std::move twice.  What am I missing?

**Comment by igorsgcampos** [comment]
Hello, I understand this is a resolved issue, but why hasn't it been merged to the arduplane-release? Every new release I have to compile the firmware myself to make the rangefinder work on my plane (I use it for rev thrust landings on a pixhawk 2 FC).

**Comment by rmackay9** [comment]
@igorsgcampos,

I can't answer directly but there's normally a lag between when it goes into master and when it's finally released into a stable build.  Pushing out a stable release takes a significant amount of time and effort for the release manager (in this case Tridge) and all the beta testers to ensure it's as close as possible to bug free.

**Pull request merged**